ParabolInc/parabol
# Update estimates when querying issues from GitHub

fix #5433

Approach:

- Fetch all labels of the GitHub issue
- For each of our estimates find corresponding labels using the current `labelTemplate` for the dimension (e.g. `Story Points: {{#}}`).
- Compare found labels values with our estimates, and insert a new row into `TaskEstimate` table if GitHub value is different
- Edge case: GitHub issue has 2 or more labels for the same dimension. Pick the first returned label that has fresh value. Remove the others to avoid the infinity loop. We can't rely on `updated_at`, it is not a date when the label was added to an issue, no such data is provided with the label by GitHub API

How to test:

- Start a poker meeting
- Connect GitHub
- Add several tasks to the scope 
- Set any estimation using a custom GitHub label e.g. `Estimate: 3`
- Open the issue on GitHub, edit the estimation, remove `Estimate: 3` and add `Estimate: 8` label
- Refresh the meeting page several times
- Check `TaskEstimate` PG table — see external estimate with value 8 added once.
- Open the issue on GitHub, add `Estimate: 3` label and **do not remove** existing `Estimate: 8` label
- Refresh the meeting page several times
- Open the issue on GitHub and see `Estimate: 8` label was removed
- Check `TaskEstimate` PG table — see external estimate with value 3 added once.
